core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -66,9 +70,20 @@ class AdminClient(val time: Time,
     throw new RuntimeException(s"Request $api failed on brokers $bootstrapBrokers")
   }
 
-  def findCoordinator(groupId: String): Node = {
+  def findCoordinator(groupId: String, timeoutMs: Long = 0,
+                      retryIntervalMs: Int = AdminClient.DefaultGroupQueryRetryIntervalMs): Node = {
+    var startTime = time.milliseconds
     val requestBuilder = new GroupCoordinatorRequest.Builder(groupId)
-    val response = sendAnyNode(ApiKeys.GROUP_COORDINATOR, requestBuilder).asInstanceOf[GroupCoordinatorResponse]
+    var response = sendAnyNode(ApiKeys.GROUP_COORDINATOR, requestBuilder).asInstanceOf[GroupCoordinatorResponse]
+
+    while (response.error == Errors.GROUP_COORDINATOR_NOT_AVAILABLE && time.milliseconds - startTime < timeoutMs) {
+      Thread.sleep(retryIntervalMs)
+      response = sendAnyNode(ApiKeys.GROUP_COORDINATOR, requestBuilder).asInstanceOf[GroupCoordinatorResponse]
+    }
+
+    if (timeoutMs > 0 && time.milliseconds - startTime > timeoutMs)
```

Comment:
Should have caught this before, but we should probably handle this the same way, right?